### ISSUE TITLE
Fix flipping card when learning

### ISSUE THREAD

**Issue by ksheremet** [dasfoo/delern#1015]
**Describe the bug**
Card doesn't flip during learning

**To Reproduce**
1. Open learning of cards
1. Click on the card to flip
1. Press 'know' or 'don't know' button
1. Click on the next card to flip
1. See error

**Expected behavior**
Card flips smoothly on click

![ezgif com-optimize](https://user-images.githubusercontent.com/6387464/56291693-05795500-6126-11e9-93d0-ce1f3d80e19f.gif)

**Comment by jddeep** [comment]
@ksheremet I would like to look into this.

**Comment by jddeep** [comment]
@ksheremet ?

**Comment by ksheremet** [comment]
@jddeep OK

**Comment by jddeep** [comment]
@ksheremet This issue occurs with a particular card for the first time. Like once the first tap is done, the Animation occurs as expected. Have you got an insight on this?

**Comment by ksheremet** [comment]
@jddeep 
I haven't had a look at it yet.
Try out to find out the solution on how to fix it. 
I recommend watching these videos about Stateful and Stateless widgets:
https://www.youtube.com/watch?v=wE7khGHVkYY&list=PLjxrf2q8roU2HdJQDjJzOeO6J3FoFLWr2&index=6&t=0s
https://www.youtube.com/watch?v=AqCMFXEmf3w&list=PLjxrf2q8roU2HdJQDjJzOeO6J3FoFLWr2&index=7&t=0s

**Comment by jddeep** [comment]
@ksheremet Yes I tried to figure out. But the card is showing a weird kind of behaviour. Like on the first tap of a new card (on the front side) the backside is shown(which is as desired) BUT without the animation. And once the card has flipped once (without animation), then it behaves as expected with the animation. @ksheremet Have you noticed this?

**Comment by jddeep** [comment]
@ksheremet Have you noticed the above?

**Comment by ksheremet** [comment]
Hi @jddeep 
I had a look. The problem is with ``` bool backshown ``` and ```controller```. When the next card appears and it shows front side. ```backshown``` param is true (from the previous card). The same is with the animation controller. It has old value from the previous card. It explained in this video: https://www.youtube.com/watch?v=AqCMFXEmf3w&list=PLjxrf2q8roU2HdJQDjJzOeO6J3FoFLWr2&index=7&t=0s

Widget and State object have a different lifespan. Try out to reset ```controller``` and ```backshown``` when the new card comes.

**Comment by jddeep** [comment]
@ksheremet Please close this.
